src/NuGet.Clients/NuGet.CommandLine/MsBuildToolsetEx.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices.ComTypes;
+using Microsoft.Build.Evaluation;
+using Microsoft.VisualStudio.Setup.Configuration;
+
+namespace NuGet.CommandLine
+{
+    public class MsBuildToolsetEx: IComparable
+    {
+        private Toolset _toolset;
+        private DateTime _installDateTime;
+        private Version _version;
+
+        public MsBuildToolsetEx(Toolset toolset)
+        {
+            _toolset = toolset;
+        }
+
+        public MsBuildToolsetEx(Toolset toolset, DateTime installDateTime)
+        {
+            _toolset = toolset;
+            _installDateTime = installDateTime;
+        }
+
+        internal MsBuildToolsetEx(ISetupInstance sxsToolset)
+        {
+            _toolset = new Toolset(toolsVersion: sxsToolset.GetInstallationVersion(),
+                        toolsPath: sxsToolset.GetInstallationPath(),
+                        projectCollection: null, 
+                        msbuildOverrideTasksPath: string.Empty);
+            _installDateTime = ConvertFILETIMEToDateTime(sxsToolset.GetInstallDate());
+        }
+
+        public string ToolsVersion
+        {
+            get
+            {
+                return _toolset?.ToolsVersion;
+            }
+        }
+
+        public Version ParsedToolsVersion
+        {
+            get
+            {
+                if (_version == null)
```

Comment:
If it's never called, there's no need to initialize. But in these cases it will likely always be called as I'm ordering the set. And good call on thread safety too.